src/base_classes/active_space_solver.cc

```diff
@@ -97,16 +97,33 @@ const std::map<StateInfo, std::vector<double>>& ActiveSpaceSolver::compute_energ
         method->compute_energy();
         const auto& energies = method->energies();
         state_energies_map_[state] = energies;
+        const auto& spin2 = method->spin2();
+
+        if (spin2.size() != 0) {
+            double target_S = 0.5 * (static_cast<double>(state.multiplicity()) - 1.0);
+            for (double root_spin2 : spin2) {
+                double root_S = 0.5 * std::sqrt(1.0 + 4.0 * root_spin2) - 0.5;
+                if (std::fabs(target_S - root_S) > 0.25) {
```

Comment:
Do we want to make the S tolerance a user-defined parameter, with 0.25 as the default?

Reply:
That's a good idea. I implemented it.